zvmsdk/volumeop.py

```
@@ -833,6 +833,9 @@ def release_fcp_devices(self, assigner_id, fcp_template_id):
                         LOG.info("Unreserve fcp device %s from "
                                  "instance %s and FCP Multipath Template %s."
                                  % (fcp_ids, assigner_id, fcp_template_id))
+                        # Sync DB to update FCP state,
+                        # so that released FCPs are set to free
+                        self._sync_db_with_zvm()
```

1. move this out of the with database.get_fcp_conn() to reduce the time that hold a db connection.
2. add a try-catch to wrap the _sync_db_with_zvm, ignore any exception, just log a warning.